cmd/notary/tuf.go

```diff
@@ -270,58 +272,82 @@ func verify(cmd *cobra.Command, args []string) {
 	return
 }
 
-func passphraseRetriever() (string, error) {
-	fmt.Println("Please provide a passphrase for this root key: ")
-	var passphrase string
-	_, err := fmt.Scanln(&passphrase)
-	if err != nil {
-		return "", err
-	}
-	if len(passphrase) < 8 {
-		fmt.Println("Please use a password manager to generate and store a good random passphrase.")
-		return "", errors.New("Passphrase too short")
-	}
-	return passphrase, nil
-}
+func getNotaryPassphraseRetriever() (trustmanager.PassphraseRetriever) {
+	userEnteredTargetsSnapshotsPass := false
+	targetsSnapshotsPass := ""
 
-func getPassphrase(confirm bool) ([]byte, error) {
-	if pass := os.Getenv("NOTARY_ROOT_PASSPHRASE"); pass != "" {
-		return []byte(pass), nil
-	}
+	return func(keyID string, alias string, createNew bool, numAttempts int) (string, bool, error) {
```

Comment:
Interesting way of using a closure to keep track of state. I don't actively dislike it, but since we use a global to store `retriever` anyway, I'm not sure why this is better than just having a global structure that stores the state and replacing `retriever` with a normal function.

Keep in mind that accesses to these state variables will need to be synchronized if it's possible for the retriever to be called by multiple goroutines. I don't think this is ever an issue in the notary client (but I could be wrong). It could become an issue if similar code is used in the Docker engine, for example.